src/ts/utils/ShadowDomManager.ts

```diff
@@ -0,0 +1,124 @@
+import { DOM } from '../DOM';
+import { ShadowDomConfig } from '../UIConfig';
+import { prefixCss } from '../components/DummyComponent';
+
+/**
+ * Encapsulates Shadow DOM initialization.
+ */
+export class ShadowDomManager {
+  private static readonly DEFAULT_SHADOW_STYLESHEET_NAME = '{{FILENAME}}';
+
+  // The ShadowRoot element where the UI is rendered into
+  private shadowRoot?: ShadowRoot;
+
+  // The wrapper element in which the ShadowRoot is rendered into.
+  // This is needed to have a proper way to destroy the ShadowRoot on release.
+  private shadowHost?: DOM;
+
+  public static isShadowDomSupported(): boolean {
+    const shadowSupported =
+      typeof ShadowRoot !== 'undefined' &&
+      typeof HTMLElement !== 'undefined' &&
+      typeof HTMLElement.prototype.attachShadow === 'function';
+
+    if (!shadowSupported) {
+      console.warn('Shadow DOM is not supported in this environment. Falling back to classic UI rendering.');
+    }
+
+    return shadowSupported;
+  }
+
+  initialize(containerElement: DOM, shadowDomConfig: ShadowDomConfig) {
+    if (!ShadowDomManager.isShadowDomSupported()) {
+      return;
+    }
+
+    const shadowHost = new DOM('div', {
+      class: prefixCss('shadow-dom-container'),
+    });
+
+    shadowHost.css({
+      position: 'absolute',
+      top: '0',
+      left: '0',
+      width: '100%',
+      height: '100%',
+    });
+
+    containerElement.append(shadowHost);
+
+    this.shadowHost = shadowHost;
+    this.shadowRoot = shadowHost.get(0).attachShadow({ mode: 'open' });
+
+    this.injectStyles(shadowDomConfig);
+  }
+
+  getShadowRoot(): ShadowRoot | undefined {
+    return this.shadowRoot;
+  }
+
+  release(): void {
+    if (this.shadowHost) {
+      this.shadowHost.remove();
+      this.shadowHost = undefined;
+      this.shadowRoot = undefined;
+    }
+  }
+
+  /**
+   * Looks for existing stylesheets on the document for the given names and injects them into the ShadowRoot to include
+   * default UI styles (and custom additional styles).
+   */
+  private injectStyles(shadowDomConfig: ShadowDomConfig): void {
+    if (!this.shadowRoot) {
+      return;
+    }
+
+    const stylesheetName = shadowDomConfig.uiStylesheetName || ShadowDomManager.DEFAULT_SHADOW_STYLESHEET_NAME;
+    const mainHref = resolveStylesheetHref(stylesheetName);
+
+    if (mainHref) {
+      appendStylesheet(this.shadowRoot, mainHref);
+    } else {
+      console.warn(
+        'Shadow DOM is enabled but no stylesheet was found. ' +
+          'Provide `shadowDomCss` (URL or filename) or include bitmovinplayer-ui.css on the page.',
```

Comment:
Shouldn't we use the `{{FILENAME}}` here, too?
And `shadowDomCss` is not a valid config property.

```suggestion
          'Provide `uiStylesheetName` (URL or filename) or include {{FILENAME}} on the page.',
```

Reply:
[20fd717](https://github.com/bitmovin/bitmovin-player-ui/pull/808/commits/20fd7178f50221e475bdf626b557875cac6869b5)